diff --git a/lgi/lgi/Makefile b/lgi/lgi/Makefile
index 08419bd..c487aca 100644
--- a/lgi/lgi/Makefile
+++ b/lgi/lgi/Makefile
@@ -11,7 +11,7 @@ LUA_VERSION=5.1
 LUA_LIBDIR = $(PREFIX)/lib/lua/$(LUA_VERSION)
 LUA_SHAREDIR = $(PREFIX)/share/lua/$(LUA_VERSION)
 
-PKG_CONFIG = pkg-config
+PKG_CONFIG = $(HOST_DIR)/bin/pkg-config
 GINAME = gobject-introspection-1.0
 PKGS = $(GINAME) gmodule-2.0 libffi
 VERSION_FILE = version.lua
